src/database/sql_database.cc

```diff
@@ -1176,14 +1176,13 @@ int SQLDatabase::createContainer(int parentID, std::string name, const std::stri
             identifier("property_name"),
             identifier("property_value"),
         };
+        std::vector<std::vector<std::string>> valuesets;
+        valuesets.reserve(itemMetadata.size());
+        const std::string newIdStr = fmt::to_string(newId);
         for (auto&& [key, val] : itemMetadata) {
-            auto mvalues = std::vector {
-                fmt::to_string(newId),
-                quote(key),
-                quote(val),
-            };
-            insert(METADATA_TABLE, mfields, mvalues);
+            valuesets.push_back({ newIdStr, quote(key), quote(val) });
```

Comment:
Sounds like emplace_back should be used.

Reply:
Only if `std::vector<std::string>` had a three-argument constructor. In fact, push_back is the only way this works: https://godbolt.org/z/nEdWoWoEY

Reply:
Hmmm yeah. Only way to make it work is to use std::tuple or something. No point.

Reply:
Ideally, we'd have a compiler enforced linking between the field names and values, too. For now, at least I added an assert that assures they have the same number of elements. Since there are only two methods that even use batched insert, the effort is not worth it.